pydantic_ai_slim/pydantic_ai/models/google.py

```diff
@@ -511,7 +525,7 @@ def _process_response_from_parts(
     for part in parts:
         if part.text is not None:
             if part.thought:
-                items.append(ThinkingPart(content=part.text))
+                items.append(ThinkingPart(content=part.text, signature=_bytes_to_base64_string(part.thought_signature)))
```

Comment:
Actually, I don't think this is going to work.

The thinking signature in the [example](https://github.com/pydantic/pydantic-ai/issues/2293) (see Appendix 1) comes right _after_ the thinking part.  

So, I think support for a `signature` field would need to be added to `ToolCallPart()` and `TextPart()`?